Tests/ScopeCheck/ambiguous-operators.silt

```diff
@@ -0,0 +1,71 @@
+-- RUN: %silt --verify scopes %s
+
+module ambiguous-operators where
+
+data True : Type where
+  | tt : True
+
+data False : Type where
+
+data Bool : Type where
+  | false : Bool
+  | true  : Bool
+
+if_then_else_ : Bool -> {A : Type} -> A -> A -> A
+if true  then x else y = x
+if false then x else y = y
+
+infix 1 if_then_else_
+
+_&&_ : Bool -> Bool -> Bool
+x && y = if_then_else_ x Bool y false
```

Comment:
Any reason this call doesn’t use mixfix notation?

Reply:
Some of this tests early-exit.  I'll add more docs, but basically it's less expensive to traverse all the operators and filter them from the grammar if they don't appear in the expression than it is to build and parse the full grammar.  It's even less expensive to do no work at all if there are no operators!